File: java/pyl/33996/code/src/main/java/io/temporal/poc/sample/SampleWorker.java
package io.temporal.poc.sample;

import io.temporal.client.WorkflowClient;
import io.temporal.serviceclient.WorkflowServiceStubs;
import io.temporal.worker.Worker;
import io.temporal.worker.WorkerFactory;

public class SampleWorker {

    static final String TASK_QUEUE = "sample-queue";

    public static void main(String[] args) {
        WorkflowServiceStubs service = WorkflowServiceStubs.newLocalServiceStubs();
        WorkflowClient client = WorkflowClient.newInstance(service);

        WorkerFactory factory = WorkerFactory.newInstance(client);
        Worker worker = factory.newWorker(TASK_QUEUE);

        // parent and child workflows are both registered on the same worker
        worker.registerWorkflowImplementationTypes(
                GreetingWorkflowImpl.class, GreetingChildWorkflowImpl.class);
        worker.registerActivitiesImplementations(new GreetingActivitiesImpl());

        factory.start();
        System.out.println("Worker started, polling '" + TASK_QUEUE + "'");
    }
}

File: java/pyl/33996/code/kotlin-2.2.0/src/main/kotlin/io/temporal/poc/sample/SampleWorker.kt
package io.temporal.poc.sample

import io.temporal.client.WorkflowClient
import io.temporal.serviceclient.WorkflowServiceStubs
import io.temporal.worker.WorkerFactory

object SampleWorker {
    const val TASK_QUEUE = "sample-queue"

    @JvmStatic
    fun main(args: Array<String>) {
        val service = WorkflowServiceStubs.newLocalServiceStubs()
        val client = WorkflowClient.newInstance(service)

        val factory = WorkerFactory.newInstance(client)
        val worker = factory.newWorker(TASK_QUEUE)

        worker.registerWorkflowImplementationTypes(
            GreetingWorkflowImpl::class.java,
            GreetingChildWorkflowImpl::class.java
        )
        worker.registerActivitiesImplementations(GreetingActivitiesImpl())

        factory.start()
        println("Worker started, polling '$TASK_QUEUE'")
    }
}